egs/icsi/s5/run.sh

```diff
@@ -0,0 +1,172 @@
+
+base_mic=$(echo $mic | sed 's/[0-9]//g') # sdm, ihm or mdm
+nmics=$(echo $mic | sed 's/[a-z]//g') # e.g. 8 for mdm8.
+
+set -euo pipefail
+
+# Path where AMI gets downloaded (or where locally available):
+# Note: provde the path to a subdirectory with meeting folders (i.e. B* ones)
+ICSI_DIR=/media/drive3/corpora/meeting_speech/speech # Default
+case $(hostname -d) in
+  fit.vutbr.cz) ICSI_DIR= ;; # BUT,
+  clsp.jhu.edu) ICSI_DIR= ;; # JHU,
+  cstr.ed.ac.uk) ICSI_DIR= ;; # Edinburgh,
+esac
+
+[ ! -r data/local/lm/final_lm ] && echo "Please, run 'run_prepare_shared.sh' first!" && exit 1
+final_lm=`cat data/local/lm/final_lm`
+LM=$final_lm.pr1-7
+
+# This recipe assumes (so far) you obtained the corpus already (can do so from LDC or http://groups.inf.ed.ac.uk/ami/icsi/)
+
+if [[ "$base_mic" =~ "mdm" ]]; then
+  echo "Running multi distant channel recipe with beamforming...."
+  PROCESSED_ICSI_DIR=$ICSI_DIR/../beamformed
+  if [ $stage -le 1 ]; then
+    # for MDM data, do beamforming
+    ! hash BeamformIt && echo "Missing BeamformIt, run 'cd ../../../tools/; extras/install_beamformit.sh; cd -;'" && exit 1
+    local/icsi_beamform.sh --cmd "$train_cmd" --nj 20 $mic $ICSI_DIR $PROCESSED_ICSI_DIR
+  fi
+else
+  PROCESSED_ICSI_DIR=$ICSI_DIR
+fi
+
+# Prepare original data directories data/ihm/train_orig, etc.
+if [ $stage -le 2 ]; then
+  local/icsi_${base_mic}_data_prep.sh $PROCESSED_ICSI_DIR $mic
+  local/icsi_${base_mic}_scoring_data_prep.sh $PROCESSED_ICSI_DIR $mic dev
+  local/icsi_${base_mic}_scoring_data_prep.sh $PROCESSED_ICSI_DIR $mic eval
+fi
+
+if [ $stage -le 3 ]; then
+  for dset in train dev eval; do
+    # this splits up the speakers (which for sdm and mdm just correspond
+    # to recordings) into 30-second chunks.  It's like a very brain-dead form
+    # of diarization; we can later replace it with 'real' diarization.
+    seconds_per_spk_max=30
+    [ "$mic" == "ihm" ] && seconds_per_spk_max=120  # speaker info for ihm is real,
+                                                    # so organize into much bigger chunks.
+
+    # Note: the 30 on the next line should have been $seconds_per_spk_max
+    # (thanks: Pavel Denisov.  This is a bug but before fixing it we'd have to
+    # test the WER impact.  I suspect it will be quite small and maybe hard to
+    # measure consistently.
+    utils/data/modify_speaker_info.sh --seconds-per-spk-max 30 \
+      data/$mic/${dset}_orig data/$mic/$dset
+  done
+fi
+
+# Feature extraction,
+if [ $stage -le 4 ]; then
+  for dset in train dev eval; do
+    steps/make_mfcc.sh --nj 15 --cmd "$train_cmd" data/$mic/$dset
+    steps/compute_cmvn_stats.sh data/$mic/$dset
+    utils/fix_data_dir.sh data/$mic/$dset
+  done
+fi
+
+# monophone training
+if [ $stage -le 5 ]; then
+  # Full set 77h, reduced set 10.8h,
+  utils/subset_data_dir.sh data/$mic/train 15000 data/$mic/train_15k
+
+  steps/train_mono.sh --nj $nj --cmd "$train_cmd" \
+    data/$mic/train_15k data/lang exp/$mic/mono
```

Comment:
It depends on the task, and amount of training data, but I usually observed that training with original noisy speech is better than that with enhanced speech (or using both), when we use DNN acoustic models. Using all microphone data is also good in terms of we'll increase the amount of speech data. The chime5 recipe is similar except for this part.

Reply:
@sw005320, can you please clarify that comment above?
Were you asking for them to make some specific change in the recipe? 
Are you saying that the enhancement is not helpful for training but can still help for test?

Reply:
> Are you saying that the enhancement is not helpful for training but can still help for test?
Right.
But this needs experiments, as it is not always true. This is just a comment.